packages/library-base/src/darwin/kotlin/io/realm/kotlin/internal/platform/SystemUtils.kt

```diff
@@ -44,6 +46,33 @@ public actual fun threadId(): ULong {
 public actual fun epochInSeconds(): Long =
     NSDate().timeIntervalSince1970().toLong()
 
+/**
+ * Inspired by: https://github.com/Kotlin/kotlinx-datetime/blob/master/core/darwin/src/Converters.kt
+ * and https://github.com/Kotlin/kotlinx-datetime/blob/master/core/native/src/Instant.kt.
+ *
+ * Even though Darwin only uses millisecond precision, it is possible that [date] uses larger resolution, storing
+ * microseconds or even nanoseconds. In this case, the sub-millisecond parts of [date] are rounded to the nearest
+ * millisecond, given that they are likely to be conversion artifacts.
+ *
+ * Since internalNow() should only logically return a value after the Unix epoch, it is safe to create a RealmInstant
+ * without considering having to pass negative nanoseconds.
+ */
+@Suppress("MagicNumber")
```

Comment:
seems like the linter didn't appreciate `secs` even though it's not really constant... is there a way to bypass this?

Reply:
This is the way to bypass it. Alternatively you need to configure an exception in https://github.com/realm/realm-kotlin/blob/master/config/detekt/detekt.yml#L565